lib/dnsimple/client.rb

```diff
@@ -147,16 +151,11 @@ def execute(method, path, data, options = {})
     #
     # @param  [String] method The HTTP method
     # @param  [String] path The path, relative to {#base_url}
-    # @param  [Hash] options Query and header params for request
+    # @param  [Hash] data The body for the request
+    # @param  [Hash] options The query and header params for the request
     # @return [HTTParty::Response]
     def request(method, path, data, options = {})
-      if data.is_a?(Hash)
-        options[:query]   = data.delete(:query)   if data.key?(:query)
-        options[:headers] = data.delete(:headers) if data.key?(:headers)
-      end
-      if !data.empty?
-        options[:body] = data
-      end
+      options[:body] = data if !data.empty?
```

Comment:
Starting to have second thoughts about this. It would also require to change the default value for `attributes` is `{}` not `nil`. We would either have to change it or to keep the `!data.empty?`. (I vote for the later).